hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataMergedLogRecordReader.java

```diff
@@ -54,15 +54,15 @@ public class HoodieMetadataMergedLogRecordReader extends HoodieMergedLogRecordSc
 
   private HoodieMetadataMergedLogRecordReader(FileSystem fs, String basePath, String partitionName,
                                               List<String> logFilePaths,
-                                              Schema readerSchema, String latestInstantTime,
+                                              InternalSchema readerSchema, String latestInstantTime,
                                               Long maxMemorySizeInBytes, int bufferSize,
                                               String spillableMapBasePath,
                                               ExternalSpillableMap.DiskMapType diskMapType,
                                               boolean isBitCaskDiskMapCompressionEnabled,
                                               Option<InstantRange> instantRange, boolean allowFullScan, boolean useScanV2) {
     super(fs, basePath, logFilePaths, readerSchema, latestInstantTime, maxMemorySizeInBytes, true, false, bufferSize,
         spillableMapBasePath, instantRange, diskMapType, isBitCaskDiskMapCompressionEnabled, false, allowFullScan,
-            Option.of(partitionName), InternalSchema.getEmptyInternalSchema(), useScanV2);
+            Option.of(partitionName), useScanV2);
```

Comment:
No need to modify this class

Reply:
Agree, current view is to revert all changes not related to flink https://github.com/apache/hudi/pull/5830#issuecomment-1314919173